docs/examples/machine_translation/transformer.md

```diff
@@ -49,79 +47,75 @@ wmt_transformer_model, wmt_src_vocab, wmt_tgt_vocab = \
                               pretrained=True,
                               ctx=ctx)
 
-print(wmt_src_vocab)
-print(wmt_tgt_vocab)
+# we are using mixed vocab of EN-DE, so the source and target langauge vocab are the same
+print(len(wmt_src_vocab), len(wmt_tgt_vocab))
 ```
 
 The Transformer model architecture is shown as below:
 
 <div style="width: 500px;">![transformer](transformer.png)</div>
 
-```{.python .input}
-print(wmt_transformer_model)
-```
-
-### Load  and Preprocess WMT 2014 Dataset
+### Load and Preprocess WMT 2014 Dataset
 
-We then load the WMT 2014 English-German test dataset for evaluation purpose.
+We then load the newstest2014 segment in WMT 2014 English-German test dataset for evaluation purpose.
 
 The following shows how to process the dataset and cache the processed dataset
 for the future use. The processing steps include:
 
-* 1) clip the source and target sequences
-* 2) split the string input to a list of tokens
-* 3) map the string token into its index in the vocabulary
-* 4) append EOS token to source sentence and add BOS and EOS tokens to target sentence.
+1) clip the source and target sequences
+2) split the string input to a list of tokens
+3) map the string token into its index in the vocabulary
+4) append EOS token to source sentence and add BOS and EOS tokens to target sentence.
 
-Let's first look at the WMT 2014 corpus.
+Let's first look at the WMT 2014 corpus. GluonNLP provides [WMT2014BPE](../../api/data.rst#gluonnlp.data.WMT2014BPE)
+and [WMT2014](../../api/data.rst#gluonnlp.data.WMT2014) classes. The former contains BPE-tokenized dataset, while
+the later contains the raw text. Here, we use the former for scoring, and the later for
+demonstrating actual translation.
 
 ```{.python .input}
 import hyperparameters as hparams
 
 wmt_data_test = nlp.data.WMT2014BPE('newstest2014',
                                     src_lang=hparams.src_lang,
-                                    tgt_lang=hparams.tgt_lang,
-                                    full=False)
+                                    tgt_lang=hparams.tgt_lang)
 print('Source language %s, Target language %s' % (hparams.src_lang, hparams.tgt_lang))
+print('Sample: "{}"'.format(wmt_data_test[0]))
```

Comment:
change to `print('Sample BPE-based Text: "{}"'.format(wmt_data_test[0]))` could be more user-friendly?